pkg/controller/job/job_controller.go

```diff
@@ -1361,18 +1361,33 @@ func jobSuspended(job *batch.Job) bool {
 	return job.Spec.Suspend != nil && *job.Spec.Suspend
 }
 
+// Wait for Pods to terminate
+func waitForPodsToTerminate(job *batch.Job) bool {
+	if job.Spec.PodFailurePolicy == nil {
+		return false
+	}
+	return job.Spec.PodFailurePolicy.DelayPodRecreationUntilTerminal != nil && *job.Spec.PodFailurePolicy.DelayPodRecreationUntilTerminal
+}
+
 // manageJob is the core method responsible for managing the number of running
 // pods according to what is specified in the job.Spec.
 // Respects back-off; does not create new pods if the back-off time has not passed
 // Does NOT modify <activePods>.
-func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, activePods []*v1.Pod, succeeded int32, succeededIndexes []interval, backoff backoffRecord) (int32, string, error) {
+func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, allPods []*v1.Pod, activePods []*v1.Pod, succeeded int32, succeededIndexes []interval, backoff backoffRecord) (int32, string, error) {
 	active := int32(len(activePods))
 	parallelism := *job.Spec.Parallelism
 	jobKey, err := controller.KeyFunc(job)
 	if err != nil {
 		utilruntime.HandleError(fmt.Errorf("Couldn't get key for job %#v: %v", job, err))
 		return 0, metrics.JobSyncActionTracking, nil
 	}
+	// Let's wait for the deletion to be successful before starting any new pods
+	// activePods filters out the terminating pods so we just check if there is a
+	// mismatch between the pod list and the active ones.
+	if waitForPodsToTerminate(job) && len(allPods) != len(activePods) {
+		klog.V(2).InfoS("Detected terminating pods so will not skip until termination", "allPods", len(allPods), "activePods", len(activePods))
+		return 0, metrics.JobSyncActionTracking, nil
```

Comment:
Not really sure what we should do with a return here.  I figured this means that we are BAU so I return JobSyncActionTracking.  But not sure if we want a new metric for this?